### PR TITLE
[T3A4][T17-A2]Clara Tan

### PR DESCRIPTION
**Pull request by hotchocolatemarshmallow**
Ready for Review

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -107,7 +114,9 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            if (!storage.save(addressBook)){
```
Comment (pixelducky) [inline src/seedu/addressbook/Main.java]: there shouldn't be a need to check the storage here. it should be as simple as a class method call. Let the catch block handle the exceptions

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -107,7 +114,9 @@ private CommandResult executeCommand(Command command)  {
...
+            	throw new FileDeletedException ("File deleted during runtime");
```
Comment (pixelducky) [inline src/seedu/addressbook/Main.java]: This exception should be thrown from StorageFile class

File: src/seedu/addressbook/storage/StorageFile.java [hunk 3]
```diff
@@ -20,129 +19,145 @@
...
-            marshaller.marshal(toSave, fileWriter);
-
-        } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
-        } catch (JAXBException jaxbe) {
-            throw new StorageOperationException("Error converting address book into storage format");
-        }
+    public boolean save(AddressBook addressBook) throws StorageOperationException {
+
+	/*
```
Comment (pixelducky) [inline src/seedu/addressbook/storage/StorageFile.java]: The `FileDeletedException` should be thrown from here instead. You should check if the file exists here. If not found, throw the exception to let it bubble up to the caller of this method.

**Comment by pixelducky** [comment]
@hotchocolatemarshmallow Good effort in implementing exceptions. Please close the PR after you have read the comments.

**Comment by hotchocolatemarshmallow** [comment]
Thanks for the review :))

**Pull request closed without merging**